### PR TITLE
Fixes #31857 - Move login-page caption to footer

### PR DESCRIPTION
**Pull request by Ron-Lavi**
The caption text, which is derived from the login footer text setting
may push the login pane to the bottom of the page:

Screenshot (55)

The small refactor moves it back to the footer where it used to live in the older login page:

Screenshot (56)

Screenshot (57)

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
Ack from @MariSvirik about the design :D

**Comment by Ron-Lavi** [comment]
rebased, @MariaAga @amirfefer can you take a look please? thanks

File: webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  if (caption.length > 500) {
+    loginCaption = (
+      <React.Fragment>
+        {showLess ? `${caption.substring(0, 500)}...` : caption}
+        <Button
+          className="btn-expand"
+          variant="link"
+          onClick={() => setShowLess(!showLess)}
+        >
+          {__('Read')} {showLess ? __('more') : __('less')}
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js]: Can this be `__('Read more')` and `__('Read less')` in case other languages have different structure for this?

File: webpack/assets/javascripts/react_app/components/LoginPage/LoginPage.scss [hunk 2]
```diff
@@ -14,6 +14,25 @@ $background_image: linear-gradient(
...
+    bottom: 0;
+    padding: 30px;
+    color: white;
+    font-size: 14px;
+
+    .btn-expand {
+      font-size: inherit;
+      padding: 0;
+      margin-left: 10px;
+      color: #52b0fb;
```
Comment (MariaAga) [inline webpack/assets/javascripts/react_app/components/LoginPage/LoginPage.scss]: Can you use a specif value from pf4 variables? or set a new color in `webpack/assets/javascripts/react_app/common/colors.scss` ?

**Review by MariaAga — CHANGES_REQUESTED** [review]
Tested and it looks good!
I'm thinking maybe all the Login components should be in the same folder? but it's out of this PRs scope probably :)
left a few comment

**Review by MariaAga — DISMISSED** [review]

File: webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../common/I18n';
+
+const LoginPageFooter = ({ caption }) => {
+  const [showLess, setShowLess] = React.useState(true);
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js]: nit: could you please import `useState` directly?

File: webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  if (!caption) return null;
+  let loginCaption = caption;
+
+  if (caption.length > 500) {
+    loginCaption = (
+      <React.Fragment>
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js]: nit: we can use the short fragment syntax -  `<>` 
```js
<>
  <ComponentA />
  <ComponentB />
</>
```

File: webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        {showLess ? `${caption.substring(0, 500)}...` : caption}
+        <Button
+          className="btn-expand"
+          variant="link"
+          onClick={() => setShowLess(!showLess)}
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/LoginPage/components/LoginPageFooter.js]: state setters functions are async (and are not activated immediately), therefore we can't trust the current state value.
This approach is safer: 
```js
setShowLess(prevState => !prevState)
```

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @laviro 👍 
I left a few comments, please also add a test for `LoginPageFooter`

**Comment by ekohl** [comment]
> The caption text, which is derived from the login footer text setting
> may push the login pane to the bottom of the page:

You can include this in the commit message as well. That makes the lives of maintainers easier.

**Comment by Ron-Lavi** [comment]
Thanks @amirfefer @ekohl! I think I addressed your suggestions :)

File: webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import { Button } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../common/I18n';
+
+const LoginPageFooter = ({ caption }) => {
+  const [showLess, setShowLess] = useState(true);
+  if (!caption) return null;
+  let loginCaption = <p>{caption}</p>;
+
+  if (caption.length > 500) {
```
Comment (ofedoren) [inline webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Just nit that can be ignored :D

Isn't it better to store such value somewhere in a `constants.js`? Maybe even in a file of general constants, so it can be reused somewhere else?

File: webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    loginCaption = (
+      <>
+        <p>{showLess ? `${caption.substring(0, 500)}...` : caption}</p>
```
Comment (ofedoren) [inline webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: ^^ here for example

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: :+1: for having constants. Back in university my programming professor would fail us on this.

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Sure, I will add it :)
Do you have a good name for it?
how about `LONG_TEXT_BREAK_CHAR`?

Comment (ofedoren) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Yeah, the naming is the hardest part :D

What about `MAX_LONG_TEXT`?

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: feels a bit too generic, @Mariaaga @amirfefer @ekohl any ideas? :D

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Honestly I'm not even convinced the show more/show less is needed at all. There appears to be plenty of space so why do we need to do it? Users can scroll if needed. I'd vote for removal and always show the full text.

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Good question @ekohl, I will speak with @marisvirik and @rohoover about it,
let's see what they think...

some of the text can be missed if the user didn't scroll, and a button might be more clear... though I am not against your idea :)

Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: @laviro @ekohl If there are no law requirements to show the entire text at once, I would stick with Read more option even though we have enough space to show it. 
1. Keeping the entire text visible would make the Login page look crowded and busy. 
2. As a user I would be confused if it's important for me to read the text in order not to miss any essential information for my next actions. Fistly, as a user, I would find the text non-essential because of its tiny font size and location in the footer. On the other hand, it's too long to just ignore it. So I believe it could create some kind of psychological distress on the page where we want users just to focus on logging in.

In conclusion, my two cents are that we don't want to overwhelm the users on the very first page with a long text area with minimum font size, which is not essential for their next actions. So keep Read more truncation.

But we can still work on the content of the actual text with the Content team and create a text variant when the Read more want be necessary and the text won't be too long.

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: I don't understand this. By default there is no text at all. There is a setting that is the login text but the admin controls this value. That means that any text that is there should be deemed important to an admin, right? The admin may even be annoyed by the Read More button.

I remain unconvinced that there's positive value in a Read More button for users and for us as maintainers there's a negative value.

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: the `Read more` will be visible only if the caption length is more than 500 chars long,
I am not sure how long users usually write those texts, maybe we should make some limit to this setting,
so we will know for sure that the users will write a max of 1000 chars for example, so we can adjust the page,
with 1000 chars I don't think we need the Read More/Less.

btw Twitter uses 280 char limit :P

Comment (MariSvirik) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: @ekohl I apologize, I got confused about the main goal of the text (its creation, etc.), I thought it's just some kind of a prefabricated legal disclaimer. Thank you @laviro for an explanation. My proposal is to keep it this way until we get some feedback from customers so our decisions are based on data, not our assumptions.

Comment (ekohl) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: I think we should drop `Read More` until there's a need for it. We never had it. That would keep this PR scoped to just "move" and not "move + add max char limit". Combining would also not be clear for users who read the changelog and thus be a surprise for upgrading users.

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/LoginPage/LoginPageFooter.js]: Sounds good, if we'll need it, we can add it in another PR..

**Review by ekohl — COMMENTED** [review]
From a change perspective :+1: but I can't really review the Javascript code.

**Review by amirfefer — APPROVED** [review]
Thanks @laviro 👍
